raster/apply-mosaic-rule-to-rasters/src/main/java/com/esri/samples/apply_mosaic_rule_to_rasters/ApplyMosaicRuleToRastersSample.java

```diff
@@ -0,0 +1,212 @@
+import javafx.scene.paint.Color;
+import javafx.scene.control.Label;
+
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment;
+import com.esri.arcgisruntime.layers.RasterLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.BasemapStyle;
+import com.esri.arcgisruntime.mapping.view.DrawStatus;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.raster.ImageServiceRaster;
+import com.esri.arcgisruntime.raster.MosaicRule;
+import com.esri.arcgisruntime.raster.MosaicMethod;
+import com.esri.arcgisruntime.raster.MosaicOperation;
+
+public class ApplyMosaicRuleToRastersSample extends Application {
+
+  private MapView mapView;
+  private MosaicRule mosaicRule;
+  private RasterLayer rasterLayer; // keep loadable in scope to avoid garbage collection
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Apply Mosaic Rule To Rasters Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a label
+      Label mosaicRuleLabel = new Label("Choose a mosaic rule: ");
+      mosaicRuleLabel.setTextFill(Color.WHITE);
+
+      // create a combo box
+      ComboBox<String> comboBox = new ComboBox<>();
+      comboBox.setMaxWidth(Double.MAX_VALUE);
+
+      // set up the control panel UI
+      VBox controlsVBox = new VBox(6);
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0, 0, 0, 0.3)"),
+        CornerRadii.EMPTY, Insets.EMPTY)));
+      controlsVBox.setPadding(new Insets(10.0));
+      controlsVBox.setMaxSize(210, 50);
+      controlsVBox.setVisible(false);
+      // add the label and combo box to the control panel
+      controlsVBox.getChildren().addAll(mosaicRuleLabel, comboBox);
+
+      // create a progress indicator
+      ProgressIndicator progressIndicator = new ProgressIndicator(ProgressIndicator.INDETERMINATE_PROGRESS);
+
+      // create a map view
+      mapView = new MapView();
+
+      // add draw status listener to the map view
+      mapView.addDrawStatusChangedListener (drawStatusChangedEvent -> {
+        // show progress indicator while map view is drawing
+        if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.IN_PROGRESS) {
+          progressIndicator.setVisible(true);
+        }
+        else {
+          progressIndicator.setVisible(false);
+        }
+      });
```

Comment:
We can simplify this a fair bit by passing the result of the comparison `==` straight into `.setVisible`.
Up to you and @Rachael-E to decide if this is a sacrifice of legibility though

Other than that, I think the inside comment does explain the whole block, so we could get rid of `// add draw status lisener...`
```suggestion
      // show progress indicator while map view is drawing
      mapView.addDrawStatusChangedListener(drawStatusChangedEvent -> 
        progressIndicator.setVisible(drawStatusChangedEvent.getDrawStatus() == DrawStatus.IN_PROGRESS)
      );
```